dlpar/dlpar_workload_setup.py

```diff
@@ -24,35 +24,184 @@
+        cmd = '/usr/lpp/htx/etc/scripts/htxd_shutdown'
+        self.shut_down_cmd = self.cmd_builder(
+            self.linux_1_pw, self.primary_lpar, cmd, flag)
+        process.system(self.shut_down_cmd, ignore_status=True)
+        time.sleep(5)
+        cmd = 'umount /htx_pmem*'
+        self.umount_htx_cmd = self.cmd_builder(
+            self.linux_1_pw, self.primary_lpar, cmd, flag)
+        process.system(self.umount_htx_cmd, shell=True, ignore_status=True)
+
+    def get_dist_name_version(self):
+        flag = "HTX"
+        self.distro_name = 'cat /etc/os-release | grep -w ID'
+        self.distro_version = 'cat /etc/os-release | grep -w VERSION_ID'
+        self.distro_name_cmd = self.cmd_builder(
+            self.linux_1_pw, self.primary_lpar, self.distro_name, flag)
+        self.distro_version_cmd = self.cmd_builder(
+            self.linux_1_pw, self.primary_lpar, self.distro_version, flag)
+        result = process.run(self.distro_name_cmd, shell=True)
+        self.get_distro_name = result.stdout.strip(
+        ).decode().replace('"', '').split("=")[-1]
+        result = process.run(self.distro_version_cmd, shell=True)
+        self.get_distro_version = result.stdout.strip().decode().replace(
+            '"', '').split("=")[-1].split(".")[0]
+
+    def install_latest_htx_rpm(self):
+        # Getting latest HTX rpm link for suse or rhel
+        distro_pattern = "%s%s" % (
+            self.get_distro_name, self.get_distro_version)
+        temp_string = process.getoutput(
+            "curl --silent %s" % (rpm_link),
+            verbose=False, shell=True, ignore_status=True)
+        matching_htx_versions = re.findall(
+            r"(?<=\>)htx\w*[-]\d*[-]\w*[.]\w*[.]\w*", str(temp_string))
+        distro_specific_htx_versions = [
+            htx_rpm for htx_rpm in matching_htx_versions
+            if distro_pattern in htx_rpm]
+        distro_specific_htx_versions.sort(reverse=True)
+        self.latest_htx_rpm = distro_specific_htx_versions[0]
+        self.rpm = 'rpm -ivh --nodeps %s%s --force' % (
+            rpm_link, self.latest_htx_rpm)
+
+        # Create a command to start the HTX on LPAR
+        self.htx_run_1_cmd = self.cmd_builder(
+            self.linux_1_pw, self.primary_lpar, self.rpm)
+        self.start_htx = '/usr/lpp/htx/etc/scripts/htxd_run'
+        self.start_htx_cmd = self.cmd_builder(
+            self.linux_1_pw, self.primary_lpar, self.start_htx)
+        self.htx_mdt_file = 'htxcmdline -createmdt'
+        self.htx_mdt_file = self.cmd_builder(
+            self.linux_1_pw, self.primary_lpar, self.start_htx)
+        if process.system(self.htx_run_1_cmd, shell=True, ignore_status=True):
+            self.log.error("Installing rpm failed")
+        time.sleep(5)
+        self.log.info("Starting the HTX Deamon")
+        process.run(self.start_htx_cmd)
+        time.sleep(5)
+        self.log.info("Creating the HTX mdt files")
+        process.run(self.htx_mdt_file)
+        time.sleep(5)
+
```

Comment:
All/most of this code is already part of HTX test. It shouldn't be duplicated here.

Reply:
> > All/most of this code is already part of HTX test. It shouldn't be duplicated here.
> 
> Existing HTX test cases can be used to set up local systems with HTX workloads. However, in order to prepare remote machines, we must run the appropriate commands on those remote machines. Before executing the DLPAR commands on the remote machine, we need to prepare the ssh command to execute on the remote machine. For more information, please refer to the comment link below.
> 
> [#2480 (comment)](https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2480#discussion_r1027755939)

Why can't this be done outside of the test case by executing HTX test case on the remote system using a cfg file (as mentioned by @abdhaleegit )

Reply:
not sure maybe we need to enhance the wrapper to achieve this as fvt-wrapper run on SUT